lib/services/git-service.ts

```diff
@@ -0,0 +1,98 @@
+import util from "../shared/util";
+
+export class GitService implements IGitService {
+    getPackageJsonFromSrc(templateName: string): Promise<any> {
+        let content: any;
+
+        return util.request({
+            method: "GET",
+            uri: util.format("https://raw.githubusercontent.com/NativeScript/%s/master/package.json", templateName),
+            json: true,
+            resolveWithFullResponse: true,
+            headers: util.defaultHeaders
+        })
+            .then((response: any) => {
+                content = response.body;
+                if (content.hasOwnProperty("templateType")) {
+                    return content;
+                }
+            })
+            .catch((error: any) => {
+                return {
+                    message: "Error retrieving " + templateName + " package.json from src",
+                    err: error
+                };
+            });
+    }
+
+    getAssetsContent(templateName: string): Promise<any> {
+        const platforms: any = {};
+
+        return util.request({
+            method: "GET",
+            uri: util.format("https://api.github.com/repos/NativeScript/%s/contents/tools/assets", templateName),
+            json: true,
+            resolveWithFullResponse: true,
+            headers: util.defaultHeaders
+        })
+            .then((response: any) => {
+                response.body.forEach((element: any) => {
+                    if (element.name.indexOf("phone") === -1) {
+                        let platform = element.name.split("-").pop().split(".").shift();
+                        if (platform !== "android" || platform !== "ios") {
+                            const rep = platform.match(/^(?!android|ios).*$/g);
+                            platform = platform.replace(rep, "thumbnail");
+                        }
+
+                        platforms[platform] = element.name;
+                    }
+                });
+
+                return this.getResourcesFromSrc(templateName, platforms);
+            })
+            .catch((error: any) => {
+                return { message: "Error retrieving assets from repository", error };
+            });
+    }
+
+    private getResourcesFromSrc(templateName: string, asset: any) {
```

Comment:
`asset` paramater name suggests something singular (at least to me) -- consider renaming this to assets or assetDictionary?